src/components/BulkEnrollmentResultsDownloadPage/index.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Redirect, useParams } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { logError } from '@edx/frontend-platform/logging';
+import { camelCaseObject } from '@edx/frontend-platform';
+import { ToastsContext } from '../Toasts';
+import EnterpriseAppSkeleton from '../EnterpriseApp/EnterpriseAppSkeleton';
+import LicenseManagerApiService from '../../data/services/LicenseManagerAPIService';
+
+const BulkEnrollmentResultsDownloadPage = ({ enterpriseId }) => {
+  const { enterpriseSlug, bulkEnrollmentJobId } = useParams();
+  const { addToast } = useContext(ToastsContext);
+  const [isLoading, setLoading] = useState(true);
+  const [redirectUrl, setRedirectUrl] = useState(null);
+  const [error, setError] = useState(null);
+  const [notReady, setNotReady] = useState(false);
+
+  useEffect(() => {
+    if (isLoading) {
+      LicenseManagerApiService.fetchBulkEnrollmentJob(enterpriseId, bulkEnrollmentJobId)
+        .then((response) => {
+          const result = camelCaseObject(response.data);
+          if (result.downloadUrl) {
+            setRedirectUrl(result.downloadUrl);
+          } else {
+            setNotReady(true);
+          }
+        })
+        .catch((err) => {
+          logError(err);
+          setError(err);
+        })
+        .finally(() => {
+          setLoading(false);
+        });
+    }
+  });
+
+  if (isLoading) {
+    return <EnterpriseAppSkeleton />;
+  }
+  if (notReady) {
+    addToast('Your download is not ready yet.');
+    return <Redirect to={`/${enterpriseSlug}/admin/learners`} />;
+  }
+  if (error) {
+    addToast('There was a problem with your request.');
+    return <Redirect to={`/${enterpriseSlug}/admin/learners`} />;
+  }
+  window.location.href = redirectUrl;
```

Comment:
this this is entirely sufficient if it works great, just curious if using https://v5.reactrouter.com/web/api/Redirect/to-string here will make testing easier (if it even works). Otherwise, your test can just mock out window.location.href call somehow (with may be weird stuff like this: https://remarkablemark.org/blog/2021/04/14/jest-mock-window-location-href/ )

Reply:
Unfortunately, in my testing/research I found that the react `Redirect` doesn't let you redirect off to a different app/domain - in this case I am sending the user to a pre-signed S3 link.